plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeWriter.java

```diff
@@ -196,31 +189,27 @@ public DataFileInfo getDataFileInfo()
                 creationTime,
                 dataFileType,
                 partitionValues,
-                readStatistics(inputFile, dataColumnTypes, rowCount));
+                readStatistics(fileWriter, rootTableLocation.appendPath(relativeFilePath), dataColumnTypes, rowCount));
     }
 
-    private static DeltaLakeJsonFileStatistics readStatistics(TrinoInputFile inputFile, Map</* lowercase */ String, Type> typeForColumn, long rowCount)
+    private static DeltaLakeJsonFileStatistics readStatistics(ParquetFileWriter fileWriter, Location path, Map</* lowercase */ String, Type> typeForColumn, long rowCount)
```

Comment:
Instead of passing `fileWriter` , please pass `FileMetaData fileMetaData` (`fileWriter.getFileMetadata()` ) to the method.
We do _read_Statistics in this method.